docs/misc/commands.md

```diff
@@ -0,0 +1,211 @@
+# Commands
+A command is a textual instruction for Minecraft to do somethings.
+See also [Commands](https://minecraft.wiki/w/commands) in the Minecraft Wiki.
+
+Commands are very useful because it provides a way for players to test
+your mod faster and cheat when he doesn't want to spend a long time to do
+not instresting things.
+
+Mojang provides a complete command system in the `com.mojang.brigadier` package, which includes
+ - Command parsing and handling
+ - Primitive [argument types](#argument-types)
+ - [Command Contexts](#command-contexts)
+ - Suggestions
+ - [Exception handling](#exceptions)
+ - Some other small things.
+
+ In fact, Brigadier is the command API of Minecraft.
+
+## Commands architecture
+Here are the main classes and concepts that command creation has.
+
+### Command Dispatcher
+The `CommandDispatcher<S>` (`S` is often `CommandSourceStack` in this case) work like a [`DeferredRegister`](../concepts/registries.md), but it's internally created.
+Its function `register` require a `LiteralArgumentBuilder` argument.
+
+### Argument Builder
+The `ArgumentBuilder` class is abstract, so it can't be used on its own.
+It has two children:
+ - `LiteralArgumentBuilder`: `CommandDispatcher#register` only accepts it. Returned after literal arguments (one-possiblity) and executions
```

Comment:
This does not explain what a literal argument is. It can be used anywhere as part of the command, though it is typically as the root a the initial sublevels.